### PR TITLE
TagBox: Add role, aria-roledescription, aria-labelledby on root and add aria-activedescendant on input

### PR DESCRIPTION
**Pull request by marker-dao**

### CODE REVIEW

File: js/ui/tag_box.js [hunk 1]
```diff
@@ -492,11 +500,34 @@ const TagBox = SelectBox.inherit({
...
+        if(!id) {
+            newValue = undefined;
+        } else if(isRemoved) {
+            newValue = value?.split(' ').filter(item => item !== id).join(' ');
+        } else {
+            newValue = [value, id].filter(Boolean).join(' ');
+        }
+
+        this.setAria('labelledby', newValue || undefined, this.$element());
+    },
```
Comment (ksercs) [inline js/ui/tag_box.js:546]:
```javascript
    _getNewLabelId(actualId, newId, shouldRemove) {
        if(!actualId) {
            return newId;
        }

        if(shouldRemove) {
            if(actualId === newId) {
                return undefined;
            }

            return actualId
                .split(' ')
                .filter(id => id !== newId)
                .join(' ');
        }

        return `${actualId} ${newId}`;
    },

    _updateElementAria(id, shouldRemove) {
        const shouldClearLabel = !id;
        if(shouldClearLabel) {
            this.setAria('label', undefined, this.$element());
            return;
        }

        const labelId = this.$element().attr('aria-labelledby');
        const newLabelId = this._getNewLabelId(labelId, id, shouldRemove);

        this.setAria('labelledby', newLabelId, this.$element());
    },
```

File: testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js [hunk 2]
```diff
@@ -7649,16 +7649,108 @@ QUnit.module('accessibility', () => {
...
 
     QUnit.test('select should have a correct aria-label', function(assert) {
         const $tagBox = $('#tagBox').dxTagBox();
         const $select = $tagBox.find(TAGBOX_SELECT_SELECTOR);
 
-        assert.equal($select.attr('aria-label'), 'Selected items', 'aria-label is correct');
+        assert.strictEqual($select.attr('aria-label'), 'Selected items', 'aria-label is correct');
+    });
+
+    QUnit.test('TagBox element should have a correctly aria attributes', function(assert) {
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js]:
```suggestion
    QUnit.test('TagBox element should have correct aria attributes', function(assert) {
```

File: testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js [hunk 3]
```diff
@@ -7649,16 +7649,108 @@ QUnit.module('accessibility', () => {
...
+        keyboard
+            .press('right')
+            .press('backspace');
+
+        const tagId = $tagBox.find(`.${TAGBOX_TAG_CLASS}`).attr('id');
+
+        assert.strictEqual($tagBox.attr('aria-labelledby'), tagId, 'aria-labelledby is set correctly');
+    });
+
+    QUnit.test('input should have aria-activedescendant with correct id if tag focused', function(assert) {
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js]:
```suggestion
    QUnit.test('input should have aria-activedescendant with correct id if tag is focused', function(assert) {
```

File: testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js [hunk 4]
```diff
@@ -7649,16 +7649,108 @@ QUnit.module('accessibility', () => {
...
+        const items = [1, 2];
+        const $tagBox = $('#tagBox').dxTagBox({
+            items,
+            value: [items[0], items[1]],
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js]:
```suggestion
            value: items,
```

File: js/ui/tag_box.js (hunk 1 above)
Comment (marker-dao) [reply js/ui/tag_box.js:546]: Nice work! I have added it here: [cb23b9cf53d35c71c0ccefc194a29f6582558815].

File: testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js (hunk 2 above)
Comment (marker-dao) [reply testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js]: Fixed here: [cb23b9cf53d35c71c0ccefc194a29f6582558815].

File: testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js (hunk 3 above)
Comment (marker-dao) [reply testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js]: Fixed here: [cb23b9cf53d35c71c0ccefc194a29f6582558815].

File: testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js (hunk 4 above)
Comment (marker-dao) [reply testing/tests/DevExpress.ui.widgets.editors/tagBox.tests.js]: Fixed here: [cb23b9cf53d35c71c0ccefc194a29f6582558815].

**Review by ksercs — APPROVED** [review]

**Review by EugeniyKiyashko — APPROVED** [review]

**Pull request merged**